src/main/webapp/blur.js

```diff
@@ -72,3 +73,42 @@ function getAverageColor(image) {
 
   return rgbToHex(...averageColorRgb);
 }
+
+/**
+ * Function to get image with some areas blurred.
+ * @param {Array<Rect>}rectsToBlur Areas to blur.
+ * @param {Image} image
+ * @param {Number} blurRadius
+ * @return {HTMLCanvasElement}
+ */
+function getImageWithBlurredAreas(rectsToBlur, image, blurRadius) {
+  // Create canvas and put original image on it.
+  const withoutBlurAreasCanvas = createCanvasForImage(image);
+  const withoutBlurAreasCtx = withoutBlurAreasCanvas.getContext('2d');
+
+  withoutBlurAreasCtx.drawImage(image, 0, 0);
+
+  // For smoothing edges of blurred areas.
+  withoutBlurAreasCtx.globalCompositeOperation = 'destination-out';
+  withoutBlurAreasCtx.filter = `blur(${blurRadius / 2}px)`; // smoothing radius
```

Comment:
Why exactly is the smoothing radius blurRadius / 2?

Reply:
I created another document explaining it :)
https://docs.google.com/document/d/1SpO85BDcJKVHaouZgMluValzAdisEvLbvYHNqNfvoB8/edit#

Reply:
Great, thank you!